src/renderer/modals/BlacklistToken/index.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import React, { useCallback } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import ModalBody from "~/renderer/components/Modal/ModalBody";
+import Box from "~/renderer/components/Box";
+import Button from "~/renderer/components/Button";
+import Modal from "~/renderer/components/Modal";
+import { blacklistToken } from "~/renderer/actions/settings";
+import { useDispatch } from "react-redux";
+import Text from "~/renderer/components/Text";
+
+const BlacklistTokenModal = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const confirmBlacklistToken = useCallback(
+    tokenId => {
+      dispatch(blacklistToken(tokenId));
+    },
+    [dispatch],
+  );
+
+  return (
+    <Modal
+      name="MODAL_BLACKLIST_TOKEN"
+      centered
+      render={({ data, onClose }) => (
+        <ModalBody
```

Comment:
can you try to split the component in two parts? the inside of render should better be a component itself. all the hooks can then move to that component too so you don't re-render the upper level Modal and it makes less nesting.

Reply:
sure, makes sense